d2/src/main/java/com/linkedin/d2/balancer/servers/ZooKeeperUriStoreFactory.java

```diff
@@ -0,0 +1,48 @@
+package com.linkedin.d2.balancer.servers;
+
+import com.linkedin.d2.balancer.properties.UriProperties;
+import com.linkedin.d2.balancer.properties.UriPropertiesJsonSerializer;
+import com.linkedin.d2.balancer.properties.UriPropertiesMerger;
+import com.linkedin.d2.discovery.stores.zk.ZKConnection;
+import com.linkedin.d2.discovery.stores.zk.ZooKeeperEphemeralStore;
+import com.linkedin.d2.discovery.stores.zk.ZookeeperChildFilter;
+import com.linkedin.d2.discovery.stores.zk.ZookeeperEphemeralPrefixGenerator;
+import com.linkedin.d2.discovery.stores.zk.builder.ZooKeeperEphemeralStoreBuilder;
+
+/**
+ * A factory class to create {@link ZooKeeperEphemeralStore < UriProperties >}
+ *
+ * @author Nizar Mankulangara (nmankulangara@linkedin.com)
+ */
+public class ZooKeeperUriStoreFactory implements ZooKeeperConnectionManager.ZKStoreFactory<UriProperties, ZooKeeperEphemeralStore<UriProperties>>
```

Comment:
Will this class replace the existing class ZKUriStoreFactory.java? Having 2 very similar classes looks confusing to me, if the plan is to use this class going forward, we can probably mark ZKUriStoreFactory as deprecated and add the doc to explain.
(I don't really see any non-testing usage of ZKUriStoreFactory today, looks like container creates its own impl of the factory interface)